File: src/main/java/com/ject/studytrip/image/domain/factory/ImageKeyFactory.java
package com.ject.studytrip.image.domain.factory;

import static com.ject.studytrip.image.domain.constants.ImageConstants.*;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class ImageKeyFactory {
    public static String createTmpKey(String keyPrefix, String id, String filename) {
        return TMP_PREFIX + KEY_PATTERN.formatted(keyPrefix, id, filename);
    }

    public static String toFinalKey(String tmpKey) {
        if (tmpKey == null || tmpKey.isBlank() || !tmpKey.startsWith(TMP_PREFIX)) {
            return null;
        }
        return tmpKey.substring(TMP_PREFIX.length());
    }
}

File: src/main/kotlin/com/ject/studytrip/image/domain/factory/ImageKeyFactory.kt
package com.ject.studytrip.image.domain.factory

import com.ject.studytrip.global.exception.CustomException
import com.ject.studytrip.image.domain.constants.ImageConstants
import com.ject.studytrip.image.domain.error.ImageErrorCode

object ImageKeyFactory {
    fun createTmpKey(
        keyPrefix: String?,
        id: String,
        filename: String,
    ): String = ImageConstants.TMP_PREFIX + ImageConstants.KEY_PATTERN.format(keyPrefix, id, filename)

    fun toFinalKey(tmpKey: String): String {
        if (!tmpKey.startsWith(ImageConstants.TMP_PREFIX)) {
            throw CustomException(ImageErrorCode.INVALID_IMAGE_KEY)
        }
        return tmpKey.removePrefix(ImageConstants.TMP_PREFIX)
    }
}
